packages/smooth_app/lib/generic_lib/widgets/images/smooth_images_sliver_grid.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:flutter/material.dart';
+import 'package:shimmer/shimmer.dart';
+import 'package:smooth_app/data_models/product_image_data.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/generic_lib/loading_sliver.dart';
+import 'package:smooth_app/generic_lib/widgets/images/smooth_image.dart';
+import 'package:smooth_app/generic_lib/widgets/images/smooth_images_view.dart';
+import 'package:smooth_app/generic_lib/widgets/picture_not_found.dart';
+
+/// Displays a [SliverGrid] with tiles showing the images passed
+/// via [imagesData]
+class SmoothImagesSliverGrid extends SmoothImagesView {
+  const SmoothImagesSliverGrid({
+    required super.imagesData,
+    super.onTap,
+    super.loading = false,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final List<MapEntry<ProductImageData, ImageProvider?>> imageList =
+        imagesData.entries.toList();
+
+    return SliverPadding(
+      padding: const EdgeInsets.all(MEDIUM_SPACE),
+      sliver: SliverGrid(
+        delegate: LoadingSliverChildBuilderDelegate(
+            loading: loading,
+            childCount: imageList.length,
+            loadingWidget: _buildShimmer(),
+            loadingCount: 6,
+            childBuilder: (BuildContext context, int index) {
+              final MapEntry<ProductImageData, ImageProvider<Object>?> entry =
+                  imageList[index];
+              final ImageProvider? imageProvider = entry.value;
+
+              return imageProvider == null
+                  ? const PictureNotFound()
+                  : Hero(
+                      tag: entry.key.imageUrl!,
+                      child: _ImageTile(
+                        image: imageProvider,
+                        onTap: onTap == null
+                            ? null
+                            : () => onTap!(entry.key, entry.value),
+                      ),
+                    );
+            }),
+        gridDelegate: const SliverGridDelegateWithMaxCrossAxisExtent(
+          maxCrossAxisExtent: VERY_LARGE_SPACE * 7,
+          childAspectRatio: 1.5,
```

Comment:
Please be more explicit about your choices of `VERY_LARGE_SPACE * 7` and `childAspectRatio: 1.5`